addons/mrp/models/mrp_bom.py

```diff
@@ -119,6 +119,15 @@ def onchange_routing_id(self):
         for line in self.bom_line_ids:
             line.operation_id = False
 
+    @api.model
+    def create(self, values):
+        res = super().create(values)
+        if 'product_qty' in values:
```

Comment:
you protect the access to product_qty but not to product_uom_id, but they're both required?
why not use the create multi, do the call to super, then loop over all bom and do this check accessing their values?